### DISCUSSION TITLE
gitignore shows certain files to contributors/users using flags

### DISCUSSION THREAD

**Discussion by 9662e103-129a** [community/community#27940] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello, I dont know if this is already a feature but I was thinking that there could be files only shown to contributors of a repository. say you have something that you cant show to regular users but repository contributors require to see it to work on the repository. I was thinking something like this in a gitignore file:

regular gitignore
`/netlify` 

new gitignore (with perms so that contributors can see the directory)
`/netlify --CONTRIB`
or 
`/netlify --contrib`

and gitignore (only allowing set users to view a file)
`/netlify --USER:USER_NAME`
or
`/netlify --user:USER_NAME`

e.g.
`/netlify --USER:win21H2`
or
`/netlify --user:win21H2`

**Comment by airtower-luna** [comment, marked as answer]
That is not possible. With Git you either have read access to a repository, or you don't. If you clone a repository, you get *everything* in it (unless you specifically tell Git not to fetch certain things). So, if the GitHub UI would hide certain files in public repositories people would only need to clone the repository to get around it.

If you have files in a repository you don't want to show to the public, don't make your repository public.

**Reply by 9662e103-129a** [reply]
Ah ok thanks for clarifying, I just started learning how gitignore works and stuff like that
